controllers/datadogagent/secret.go

```diff
@@ -33,17 +32,17 @@ func (r *Reconciler) manageAgentSecret(logger logr.Logger, dda *datadoghqv1alpha
 	}
 	now := metav1.NewTime(time.Now())
 	// checks token secret
-	secretName, _ := utils.GetAppKeySecret(dda)
```

Comment:
the secret name is always equal to the `DatadogAgent.Name` ?

Reply:
should we remove this function if it isn't used anymore

Reply:
In fact, `utils.GetAppKeySecret(dda)` returns the secret that holds the APP key.
So, it can be:
* either the user-supplied secret if the user has setup `credentials.appSecret` in its `datadogagent` CR,
* or `DatadogAgent.Name` if the APP key is in the operator-managed secret if the user has setup `credentials.appKey` in its `datadogagent` CR.

So, `func GetAppKeySecret(…)` is used to build the agent manifest to setup the `DD_APP_KEY` environment variable.
But in this context, here, we need the name of the operator-managed secret and we don’t care whether the APP key as been provided in clear or in a user-supplied secret.